Source/WebCore/rendering/svg/RenderSVGText.cpp

```diff
@@ -123,35 +124,32 @@ static inline void collectLayoutAttributes(RenderObject* text, Vector<SVGTextLay
     }
 }
 
-static inline bool findPreviousAndNextAttributes(RenderElement& start, RenderSVGInlineText* locateElement, bool& stopAfterNext, SVGTextLayoutAttributes*& previous, SVGTextLayoutAttributes*& next)
+static inline std::tuple<SVGTextLayoutAttributes*, SVGTextLayoutAttributes*> findPreviousAndNextAttributes(RenderElement& root, RenderSVGInlineText& locateElement)
 {
-    ASSERT(locateElement);
-    // FIXME: Make this iterative.
-    for (CheckedRef child : childrenOfType<RenderObject>(start)) {
-        if (auto* text = dynamicDowncast<RenderSVGInlineText>(child.get())) {
-            if (locateElement != text) {
-                if (stopAfterNext) {
+    SVGTextLayoutAttributes* previous = nullptr;
+    SVGTextLayoutAttributes* next = nullptr;
+    bool foundLocateElement = false;
+
+    for (CheckedPtr current = root.firstChild(); current;) {
+        if (auto* text = dynamicDowncast<RenderSVGInlineText>(*current)) {
+            if (&locateElement != text) {
+                if (foundLocateElement) {
                     next = text->layoutAttributes();
-                    return true;
+                    return { previous, next };
                 }
-
                 previous = text->layoutAttributes();
+            } else
+                foundLocateElement = true;
+        } else if (auto* childSVGInline = dynamicDowncast<RenderSVGInline>(*current)) {
```

Comment:
I think if we start by checking `RenderSVGInline` this function will be clearer:
```
for (CheckedPtr current = root.firstChild(); current;) {
    if (auto* inline = dynamicDowncast<RenderSVGInline>(*current)) {
        if (auto* child = inline->firstChild())
            current = child;
        else
            current = current->nextInPreOrderAfterChildren(&root);     
        continue;
    }

    if (auto* text = dynamicDowncast<RenderSVGInlineText>(*current)) {
        ...
    }

    current = current->nextInPreOrderAfterChildren(&root);
}